### PR TITLE
Return txt records as a dictionary instead of a string

### PR DESCRIPTION
**Pull request by makeitnew**

### CODE REVIEW

File: ios/RNZeroconf/RNNetServiceSerializer.m [hunk 1]
```diff
@@ -14,7 +14,7 @@
 const NSString *kRNServiceKeysAddresses = @"addresses";
 const NSString *kRNServiceKeysHost = @"host";
 const NSString *kRNServiceKeysPort = @"port";
-const NSString *kRNServiceTxtRecord = @"txt";
+const NSString *kRNServiceTxtRecords = @"txtRecords";
```
Comment (Apercu) [inline ios/RNZeroconf/RNNetServiceSerializer.m]: We'll try to make the api between Android and iOS compatible, so you should change it to `txt` back, even if it's now a record

File: ios/RNZeroconf/RNNetServiceSerializer.m [hunk 2]
```diff
@@ -29,8 +29,17 @@ + (NSDictionary *) serializeServiceToDictionary:(NSNetService *)service
...
+        
+        NSDictionary<NSString *, NSData *> *txtRecordDict = [NSNetService dictionaryFromTXTRecordData:service.TXTRecordData];
+        
+        NSMutableDictionary *dict = [[NSMutableDictionary alloc] init];
+        for(NSString *key in txtRecordDict) {
+            dict[key] =  [[NSString alloc]
+                          initWithData:txtRecordDict[key]
+                          encoding:NSASCIIStringEncoding];
+        }
+
```
Comment (Apercu) [inline ios/RNZeroconf/RNNetServiceSerializer.m]: remove the extra-newline here

File: ios/RNZeroconf/RNNetServiceSerializer.m [hunk 3]
```diff
@@ -29,8 +29,17 @@ + (NSDictionary *) serializeServiceToDictionary:(NSNetService *)service
         serviceInfo[kRNServiceKeysAddresses] = [self addressesFromService:service];
         serviceInfo[kRNServiceKeysHost] = service.hostName;
         serviceInfo[kRNServiceKeysPort] = @(service.port);
-        NSString* txtStr = [[NSString alloc] initWithData:service.TXTRecordData encoding:NSASCIIStringEncoding];
-        serviceInfo[kRNServiceTxtRecord] = txtStr;
+        
+        NSDictionary<NSString *, NSData *> *txtRecordDict = [NSNetService dictionaryFromTXTRecordData:service.TXTRecordData];
+        
+        NSMutableDictionary *dict = [[NSMutableDictionary alloc] init];
+        for(NSString *key in txtRecordDict) {
```
Comment (Apercu) [inline ios/RNZeroconf/RNNetServiceSerializer.m]: Missing space `for (`

File: ios/RNZeroconf/RNNetServiceSerializer.m [hunk 4]
```diff
@@ -29,8 +29,17 @@ + (NSDictionary *) serializeServiceToDictionary:(NSNetService *)service
...
+            dict[key] =  [[NSString alloc]
```
Comment (Apercu) [inline ios/RNZeroconf/RNNetServiceSerializer.m]: extra space after `=`

**Comment by makeitnew** [comment]
I made the requested changes. Let me know if I missed anything.

**Comment by Apercu** [comment]
Perfect, thanks!

**Pull request merged**